addons/sale_timesheet/models/sale_order.py

```diff
@@ -231,10 +231,15 @@ def _timesheet_compute_delivered_quantity_domain(self):
 
     def _convert_qty_company_hours(self, dest_company):
         company_time_uom_id = dest_company.project_time_mode_id
-        if self.product_uom.id != company_time_uom_id.id and self.product_uom.category_id.id == company_time_uom_id.category_id.id:
-            planned_hours = self.product_uom._compute_quantity(self.product_uom_qty, company_time_uom_id)
-        else:
-            planned_hours = self.product_uom_qty
+        planned_hours = 0.0
+        product_uom = self.product_uom
+        if product_uom == self.env.ref('uom.product_uom_unit'):
+            product_uom = self.env.ref('uom.product_uom_hour')
+        if product_uom.category_id == company_time_uom_id.category_id:
+            if product_uom != company_time_uom_id:
+                planned_hours = product_uom._compute_quantity(self.product_uom_qty, company_time_uom_id)
+            else:
+                planned_hours = self.product_uom_qty
```

Comment:
Since the `_compute_quantity` does the same check, that is
- check if the category of each uom is the same
- check if the both uoms are the same or not

I guess we could do this:
```suggestion
        planned_hours = product_uom._compute_quantity(self.product_uom_qty, company_time_uom_id, raise_if_failure=False)
```
See [here the code of that method](https://github.com/odoo/odoo/blob/ba854942355e630bc39234f2aec9c91df8f05662/addons/uom/models/uom_uom.py#L198-L226)

And so, if you agree with my suggestion, we could also remove the line:
```py
planned_hours = 0.0
```

What do you think? 🙂